apps/price_pusher/src/pyth-price-listener.ts

```diff
@@ -71,22 +90,15 @@ export class PythPriceListener implements IPriceListener {
       });
     };
 
-    eventSource.onerror = (error: Event) => {
+    eventSource.onerror = async (error: Event) => {
       console.error("Error receiving updates from Hermes:", error);
       eventSource.close();
+      await sleep(5000); // Wait a bit before trying to reconnect
+      this.startListening(); // Attempt to restart the listener
```

Comment:
please note that we have another loop checking for latest price update that crashes if there's no update within last 30 seconds.